lms/static/coffee/src/instructor_dashboard/data_download.coffee

```diff
@@ -203,6 +204,35 @@ class ReportDownloads
     grid = new Slick.Grid($table_placeholder, report_downloads_data, columns, options)
     grid.autosizeColumns()
 
+    $delete_btns = @$section.find(".delete-report")
+    $delete_btns.click (e) =>
+      table_row = jQuery(e.target.parentElement.parentElement)
+      filename_cell = table_row.find(".course-forums-data")
+      @$file_to_delete = filename_cell.text()
+      if confirm gettext 'Are you sure you want to delete the file ' + @$file_to_delete + '? This cannot be undone.'
+        @$delete_element = @$section.find ".report-downloads-delete"
+        @$delete_endpoint = @$delete_element.data 'endpoint'
+        @delete_report()
+        @remove_row_from_ui table_row
+
+  remove_row_from_ui: (row) =>
+    row_height = row.height()
+    rows_after = row.nextAll()
+    row.remove()
+    for sib_row in rows_after
```

Comment:
If you want assistance with the HTML for this @azimpradhan, don't hesitate to come ask for help from me!